### PR TITLE
add. enhancements in date-picker and TimePicker

### PR DESCRIPTION
**Pull request by abinaya-u**
Asana: https://app.asana.com/0/1204322126657135/1206514082919772/f

### CODE REVIEW

**Comment by Arjun-Go** [comment]
@abinaya-u , we also have to add a custom validation for 12 hour time picker to keep things consistent

**Comment by abinaya-u** [comment]
> @abinaya-u , we also have to add a custom validation for 12 hour time picker to keep things consistent

updated 12-hr time picker with the changes.

File: src/components/TimePicker24Hour/TimePicker24Hour.jsx [hunk 1]
```diff
@@ -39,10 +51,18 @@ const TimePicker24Hour = (props) => {
     const displayTime = e.target.value;
     const newTime = moment(displayTime, "HH:mm").format("HH:mm");
     if (isValidTime(newTime)) {
-      setWarning(false);
+      if (customValidation) {
+        customValidation(newTime);
+      } else {
+        setWarning(false);
```
Comment (ayush34957) [inline src/components/TimePicker24Hour/TimePicker24Hour.jsx:57]: I guess we need to invoke actionForInvalidTime(false) after line 57 to indicate parent that the time is valid.

File: src/components/TimePicker24Hour/TimePicker24Hour.jsx [hunk 2]
```diff
@@ -39,10 +51,18 @@ const TimePicker24Hour = (props) => {
...
+      }
       setTime(newTime);
     } else {
       setWarning(true);
+      setWarningText(
+        invalidText || "Please enter a valid time in 24-hr format"
+      );
+      actionForInvalidTime(true);
```
Comment (ayush34957) [inline src/components/TimePicker24Hour/TimePicker24Hour.jsx]: undefined check for actionForInvalidTime callback prop is missed here

Comment (abinaya-u) [reply src/components/TimePicker24Hour/TimePicker24Hour.jsx]: added undefined check

File: src/components/TimePicker/TimePicker.jsx [hunk 3]
```diff
@@ -68,11 +87,20 @@ const TimePickerCarbon = (props) => {
     const newPeriod = e.target.value;
     const selectedTime = moment(time + newPeriod, "hh:mm A").format("hh:mm A");
     if (isValidTime(selectedTime)) {
-      setWarning(false);
+      if (customValidation) {
+        customValidation(selectedTime);
+      } else {
+        setWarning(false);
+        actionForInvalidTime && actionForInvalidTime(true);
```
Comment (ayush34957) [inline src/components/TimePicker/TimePicker.jsx]: Shouldn’t we pass false in case when time is valid?

**Review by ayush34957 — APPROVED** [review]

**Pull request merged**